adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -1316,6 +1351,47 @@ public OrderData handleComponentResult(String resultJson) throws Exception {
     }
 
 
+    private OrderModel retrieveOrder(String orderCode) throws InvalidCartException {
+        OrderModel orderModel = getOrderRepository().getOrderModel(orderCode);
+        if (orderModel == null) {
+            //TODO change exception
+            throw new InvalidCartException("Order does not exist!");
+        }
+
+        getSessionService().removeAttribute(SESSION_LOCKED_CART);
+        getSessionService().removeAttribute(SESSION_PAYMENT_DATA);
+        getSessionService().removeAttribute(THREEDS2_FINGERPRINT_TOKEN);
+        getSessionService().removeAttribute(THREEDS2_CHALLENGE_TOKEN);
+        getSessionService().removeAttribute(PAYMENT_METHOD);
+
```

Comment:
do we need to remove `SESSION_MD` as well or is it not necessary/applicable?

Reply:
makes sense, will add that

Reply:
we're working on this method on a separate branch
pushed there https://github.com/Adyen/adyen-hybris/tree/PW-2530-restore-cart